File: fizz_buzz_ex/test/fizz_buzz_test.exs
defmodule FizzBuzzTest do
  use ExUnit.Case

  test "On 1 should return 1" do
  	s = FizzBuzz.process(1)
    assert s == "1"
  end

  test "On 3 should return fizz" do
    s = FizzBuzz.process(3)
    assert s == "fizz"
  end

  test "On 6 should return fizz" do
    s = FizzBuzz.process(6)
    assert s == "fizz"
  end

  test "On 5 should return buzz" do
    s = FizzBuzz.process(5)
    assert s == "buzz"
  end

  test "On 10 should return buzz" do
    s = FizzBuzz.process(10)
    assert s == "buzz"
  end

  test "On 15 should return fizzbuzz" do
    s = FizzBuzz.process(15)
    assert s == "fizzbuzz"
  end
  
end

File: fizzbuzz_cs/FizzBuzz_test.cs
using Xunit;

public class FizzBuzzTest 
{
    FizzBuzz _fb;

    public FizzBuzzTest()
    {
        _fb = new FizzBuzz();
    }

    [Theory]
    [InlineData(1)]
    [InlineData(2)]
    [InlineData(4)]
    [InlineData(7)]
    [InlineData(8)]
    public void With_N_should_return_N(int n)
    {
        var result = _fb.Parse(n);
        Assert.Equal(n.ToString(), result);
    } 

    [Theory]
    [InlineData(3)]
    [InlineData(6)]
    public void With_MultipleOf3_should_return_Fizz(int n)
    {
        var result = _fb.Parse(n);
        Assert.Equal("Fizz", result);
    }

    [Theory]
    [InlineData(5)]
    [InlineData(10)]
    public void With_MultipleOf5_should_return_Buzz(int n)
    {
        var result = _fb.Parse(n);
        Assert.Equal("Buzz", result);
    }

    [Theory]
    [InlineData(15)]
    [InlineData(30)]
    public void With_MultipleOf3And5_should_return_FizzBuzz(int n)
    {
        var result = _fb.Parse(n);
        Assert.Equal("FizzBuzz", result);
    }


}